### PR TITLE
way-cooler: Remove

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

See #77625

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @matthiasbeyer @adisbladis @grahamc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -21994,8 +21994,6 @@ in
     pythonPackages = python3Packages;
   };
 
-  way-cooler = callPackage ../applications/window-managers/way-cooler {};
```
Comment (Infinisil) [inline pkgs/top-level/all-packages.nix]: I'd like to have an error message explaining the removal here with `way-cooler = throw "..."`

File: nixos/modules/module-list.nix [hunk 2]
```diff
@@ -158,7 +158,6 @@
   ./programs/venus.nix
   ./programs/vim.nix
   ./programs/wavemon.nix
-  ./programs/way-cooler.nix
```
Comment (Infinisil) [inline nixos/modules/module-list.nix]: And also a `mkRemovedOptionModule` in `rename.nix` for the module removal

**Review by infinisil — CHANGES_REQUESTED** [review]

**Comment by gnidorah** [comment]
@Infinisil Done, thanks!

File: nixos/modules/rename.nix [hunk 3]
```diff
@@ -27,6 +27,7 @@ with lib;
     (mkRemovedOptionModule [ "services.osquery" ] "The osquery module has been removed")
     (mkRemovedOptionModule [ "services.fourStore" ] "The fourStore module has been removed")
     (mkRemovedOptionModule [ "services.fourStoreEndpoint" ] "The fourStoreEndpoint module has been removed")
+    (mkRemovedOptionModule [ "programs" "way-cooler" ] "The Way Cooler module has been removed")
```
Comment (Infinisil) [inline nixos/modules/rename.nix]: I think it wouldn't hurt to have a better error message here either

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -21994,7 +21994,8 @@ in
...
+  way-cooler = throw ("Project abandoned by its author: " +
+    "http://way-cooler.org/blog/2020/01/09/way-cooler-post-mortem.html");
```
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]:
```suggestion
    "https://way-cooler.org/blog/2020/01/09/way-cooler-post-mortem.html");
```

Https seems to work

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -21994,7 +21994,8 @@ in
     pythonPackages = python3Packages;
   };
 
-  way-cooler = callPackage ../applications/window-managers/way-cooler {};
+  way-cooler = throw ("Project abandoned by its author: " +
```
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]:
```suggestion
  way-cooler = throw ("way-cooler is abandoned by its author: " +
```

else this message just confuses people

**Comment by gnidorah** [comment]
@Infinisil @worldofpeace Fixed, thanks!

**Review by worldofpeace — APPROVED** [review]
LGTM, please squash your commits :+1:

**Comment by gnidorah** [comment]
@worldofpeace squashed!

**Comment by worldofpeace** [comment]
> @worldofpeace squashed!

Thanks for cleaning this up @gnidorah

**Comment by gnidorah** [comment]
@worldofpeace Thanks!

**Pull request merged**